src/manifolds/QuotientManifold.jl

```diff
@@ -0,0 +1,169 @@
+struct QuotientManifold{𝔽,MT<:AbstractManifold{𝔽},NT<:AbstractManifold} <:
+       AbstractDecoratorManifold{𝔽}
+    manifold::MT
+    total_space::NT
+end
+
+@inline active_traits(f, ::QuotientManifold, ::Any...) = merge_traits(IsQuotientManifold())
+
+@doc raw"""
+    canonical_project(M, p)
+
+compute the canonical projection ``π`` on a manifold ``\mathcal M`` that
+[`IsQuotientManifold`](@ref), e.g. a [`QuotientManifold`](@ref).
+The canonical (or natural) projection ``π`` from the total space ``\mathcal N``
+onto ``\mathcal M`` given by
+
+```math
+    π = π_{\mathcal N, \mathcal M} : \mathcal N → \mathcal M, p ↦ π_{\mathcal N, \mathcal M}(p) = [p].
+```
+
+in other words, this function implicitly assumes, that the total space ``\mathcal N`` is given,
+for example explicitly when `M` is a [`QuotientManifold`](@ref) and `p` is a point on `N`.
+"""
+function canonical_project(M::AbstractManifold, p)
+    q = allocate_result(M, canonical_project, p)
+    return canonical_project!(M, q, p)
+end
+
+@doc raw"""
+    canonical_project!(M, q, p)
+
+compute the canonical projection ``π`` on a manifold ``\mathcal M`` that
+[`IsQuotientManifold`](@ref), e.g. a [`QuotientManifold`](@ref) in place of `q`.
+
+See [`canonical_project`](@ref) for more details.
+"""
+canonical_project!(M::AbstractManifold, q, p)
+
+decorated_manifold(M::QuotientManifold) = M.manifold
+
+@doc raw"""
+    differential_canonical_project(M, p, X)
+
+compute the differential of the canonical projection ``π`` on a manifold ``\mathcal M`` that
+[`IsQuotientManifold`](@ref), e.g. a [`QuotientManifold`](@ref).
+The canonical (or natural) projection ``π`` from the total space ``\mathcal N``
+onto ``\mathcal M``, such that its differential
+
+```math
+ Dπ(p) : T_p\mathcal N → T_{π(p)}\mathcal M
+```
+
+where again the total space might be implicitly assumed, or explicitly when using a
+[`QuotientManifold`](@ref) `M`. So here `p` is a point on `N` and `X` is from ``T_p\mathcal N``.
+"""
+function differential_canonical_project(M::AbstractManifold, p, X)
+    q = allocate_result(M, differential_canonical_project, p, X)
+    return differential_canonical_project!(M, q, p, X)
+end
+
+@doc raw"""
+    differential_canonical_project!(M, Y, p, X)
+
+compute the differential of the canonical projection ``π`` on a manifold ``\mathcal M`` that
+[`IsQuotientManifold`](@ref), e.g. a [`QuotientManifold`](@ref). See [`differential_canonical_project`](@ref) for details.
+"""
+differential_canonical_project!(M::AbstractManifold, q, p)
+
+@doc raw"""
+    get_total_space(M::AbstractDecoratorManifold)
+
+Return the total space of a manifold that [`IsQuotientManifold`](@ref), e.g.
+a [`QuotientManifold`](@ref).
+"""
+get_total_space(::AbstractManifold)
+get_total_space(M::QuotientManifold) = M.total_space
+
+@doc raw"""
+    get_orbit_action(M::AbstractDecoratorManifold)
+
+Return the group action that generates the orbit of an equivalence class of the quotient manifold `M`,
+that is for the case that
```

Comment:
```suggestion
Return the group action that generates the orbit of an equivalence class of the quotient manifold `M`
for which equivalence classes are orbits of an action of a Lie group. For the case that
```
Just a little more details, since it's possible to define a quotient manifold without an action.